### PR TITLE
Add class-wide validation for accounts

### PR DESCRIPTION
**Pull request by lwander**
@duftler or @ttomsu or @jtk54 PTAL

### CODE REVIEW

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/model/v1/Updateable.java [hunk 1]
```diff
@@ -100,4 +105,39 @@
...
+   * Validate this entire class
+   *
+   * @return The list of errors when validation fails ([] when there are no failures and validation has passed).
+   * @throws IllegalAccessException
+   * @throws NoSuchFieldException
+   */
+  default public List<String> validate() throws IllegalAccessException, NoSuchFieldException {
+    List<String> errors = new ArrayList<>();
+
+    if (this instanceof Account) {
```
Comment (jtk54) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/model/v1/Updateable.java]: Will this only apply to `Account`s? Why is this in the interface? Seems like this ought to be in `Account`, particularly if you are planning on using `Updateable` for other things.

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/model/v1/Updateable.java [hunk 2]
```diff
@@ -100,4 +105,39 @@
...
+          .map(v -> (ValidateAccount) v)
+          .map(ValidateAccount::validators)                                      // Pick of the validators
+          .flatMap(Stream::of)                                                   // Flatten the stream of lists
+          .map(v -> {
+            try {
+              return v.getConstructor(this.getClass()).newInstance(this);        // Construct the validators
+            } catch (Exception e) {
+              throw new RuntimeException(e);
+            }
+          }).filter(v -> !v.skip())                                              // Ignore skippable validators
```
Comment (jtk54) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/model/v1/Updateable.java]: Maybe put the `.filter` on a new line for consistency. Just a nit though.

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/model/v1/Updateable.java (hunk 1 above)
Comment (lwander) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/model/v1/Updateable.java]: Good thinking

File: halyard-config/src/main/java/com/netflix/spinnaker/halyard/model/v1/Updateable.java (hunk 2 above)
Comment (lwander) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/model/v1/Updateable.java]: This is how the intellij settings shipping with gradle are formatting it, and I'd rather rely on that w/o too much thought about formatting by hand.

Comment (jtk54) [inline halyard-config/src/main/java/com/netflix/spinnaker/halyard/model/v1/Updateable.java]: That's ok with me.

**Pull request merged**